geomstats/learning/riemannian_mean_shift.py

```diff
@@ -1,147 +1,161 @@
+        All points in 'bandwidth' size around center are considered for
+        calculating new mean centers.
+    tot : float, stopping condition
+        Computation of subsequent mean centers is stopped when the distance
+        between them is less than 'tot'
+    mean : Mean method used to calculate the new center of a cluster
+    n_centers : Total initial centers
+    n_jobs : Number of parallel threads to be initiated for parallel computation
+    max_iter : int
+        Upper bound on total number of iterations for the centers to converge
+    init_centers : Initializing centers, either from the given input points or
+             random points uniformly distributed in the input manifold
+    kernel : Weighing function to assign kernel weights to each center
+    """
+
+    def __init__(
+        self,
+        manifold,
+        metric,
+        bandwidth,
+        tol=1e-2,
+        n_centers=1,
+        n_jobs=1,
+        max_iter=100,
+        init_centers="from_points",
+        kernel="flat",
+        **frechet_mean_kwargs
+    ):
         self.manifold = manifold
         self.metric = metric
         self.bandwidth = bandwidth
         self.tol = tol
-        self.mean = FrechetMean(self.metric, **FrechetMean_kwargs)
+        self.mean = FrechetMean(self.metric, **frechet_mean_kwargs)
+        self.n_centers = n_centers
+        self.n_jobs = n_jobs
+        self.max_iter = max_iter
+        self.init_centers = init_centers
+        self.kernel = kernel
         self.centers = None
 
-    def __intersets_distances(self, points_A, points_B, n_jobs=1, **joblib_kwargs):
-        """
-        Parallel computation of distances between two sets of points.
-        """
+        print(type(max_iter))
+        print(max_iter)
+
+    def __intersets_distances(self, points_A, points_B, **joblib_kwargs):
+        """Parallel computation of distances between two sets of points"""
         n_A, n_B = points_A.shape[0], points_B.shape[0]
 
         @joblib.delayed
         @joblib.wrap_non_picklable_objects
         def pickable_dist(x, y):
+            """Riemannian metric between points x & y"""
             return self.metric.dist(x, y)
 
-        pool = joblib.Parallel(n_jobs=n_jobs, **joblib_kwargs)
+        pool = joblib.Parallel(n_jobs=self.n_jobs, **joblib_kwargs)
         out = pool(
             pickable_dist(points_A[i, :], points_B[j, :])
             for i in range(n_A)
             for j in range(n_B)
         )
 
-        # assuming numpy backend
-        return np.array(out).reshape((n_A, n_B))
+        return gs.array(out).reshape((n_A, n_B))
 
-    def fit(
-        self,
-        points,
-        n_centers,
-        n_jobs=1,
-        max_iter=100,
-        init_centers="from_points",
-        kernel="flat",
-        **joblib_kwargs
-    ):
-        """
-        Fit centers in all the input points to find 'n_centers' number of clusters.
-        """
+    def fit(self, X):
```

Comment:
For compatibility with pipelines
```suggestion
    def fit(self, X, y=None):
```

Reply:
Am confused about this comment. What should I assign to y, since we are trying to fit the centers in "self.centers".

Reply:
It is for API compatibility, you can just add `, y=None` after the `X`. It will not be used.

Reply:
Okay noted.